flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionGraphHandler.java

```diff
@@ -204,7 +204,7 @@ public SerializedInputSplit requestNextInputSplit(
 
         final InputSplit nextInputSplit = execution.getNextInputSplit();
 
-        if (log.isDebugEnabled()) {
+        if (nextInputSplit != null) {
             log.debug("Send next input split {}.", nextInputSplit);
         }
```

Comment:
Where is it logged that there are no more input splits?

Reply:
`org.apache.flink.api.common.io.DefaultInputSplitAssigner#getNextInputSplit`

Maybe we should move that logging into the ExecutionGraphHandler so it applies to all implementations?

Reply:
Sounds like a good idea.